mbedtls/src/x509/certificate.rs

```diff
@@ -221,11 +221,21 @@ impl Certificate {
         MdType::from(self.inner.sig_md)
     }
 
-    pub fn verify(
+    fn verify_ex<F>(
         chain: &MbedtlsList<Certificate>,
         trust_ca: &MbedtlsList<Certificate>,
         err_info: Option<&mut String>,
-    ) -> Result<()> {
+        cb: Option<&F>,
+    ) -> Result<()>
+    where
+        F: VerifyCallback + 'static,
+    {
+        let (f_vrfy, p_vrfy): (Option<unsafe extern "C" fn(_, _, _, _) -> _>, _) = if let Some(cb) = cb.as_ref() {
```

Comment:
if you use
```
cb: Option<&F>,
```

Then you shouldn't:
```
if let Some(cb) = cb.as_ref()
```

cb would now be a Option<&&F>

if you then cast this cb via:
```
cb as *const _ as *mut c_void)
```

Do you get a pointer to a pointer, or a proper pointer?

This would be solved either if you pass cb as Option<F> or not use as_ref here?